shell/core/extension-manager-impl.js

```diff
@@ -464,6 +470,19 @@ const createExtensionManager = (context) => {
       return extensions;
     },
 
+    getHostedProviders(context) {
```

Comment:
nit: From my perspective, `extension-manager-impl.js` is best served as a manager for handling loading and registering extensions. `getHostedProviders()` & `isHostedProvider()` aren't concerned with extension management. I'm considering if there might be a better location for this functionality.. Perhaps, `shell/utils/cluster.js`?

On the other hand, it might make sense to keep this close to `getProviders()`.